src/main/groovy/com/avast/gradle/dockercompose/tasks/ComposeUp.groovy

```diff
@@ -102,13 +109,19 @@ class ComposeUp extends DefaultTask {
     }
 
     protected ServiceInfo createServiceInfo(String serviceName) {
-        String containerId = getContainerId(serviceName)
-        logger.info("Container ID of service $serviceName is $containerId")
-        def inspection = getDockerInspection(containerId)
-        ServiceHost host = getServiceHost(serviceName, inspection)
-        logger.info("Will use $host as host of service $serviceName")
-        def tcpPorts = getTcpPortsMapping(serviceName, inspection, host)
-        new ServiceInfo(name: serviceName, serviceHost: host, tcpPorts: tcpPorts, containerHostname: inspection.Config.Hostname, inspection: inspection)
+        Iterable<String> containerIds = getContainerIds(serviceName)
+        ServiceInfo serviceInfo = new ServiceInfo(name: serviceName)

```

Comment:
As the `ServiceInfo` will be immutable, you will have to move this constructor call to the end of this method.